### PR TITLE
parser: add camelcase and pascalcase-modifiers for lsp-snippets.

### PR DESCRIPTION
**Pull request by L3MON4D3**
Yeah, pretty much title.

Needs:
- [x] doc (doesn't really need any documentation)
- [x] tests

### CODE REVIEW

File: lua/luasnip/util/str.lua [hunk 1]
```diff
@@ -113,4 +113,34 @@ function M.sanitize(str)
...
+	-- uppercase first character.
+	return str:sub(1, 1):upper() .. str:sub(2, -1)
+end
+M.vscode_string_modifiers = {
+	upcase = string.upper,
+	downcase = string.lower,
+	capitalize = capitalize,
+	pascalcase = function(str)
+		local pascalcased = ""
+		for word in str:gmatch("[a-zA-Z0-9]+") do
```
Comment (leiserfg) [inline lua/luasnip/util/str.lua:129]: Is not better to use `gsub` here?

Comment (L3MON4D3) [reply lua/luasnip/util/str.lua:129]: Ouh, yeah that would be better :+1:

Comment (L3MON4D3) [reply lua/luasnip/util/str.lua]: (alternatively, remove all non`[a-zA-Z0-9]`-characters with a second gsub, but I think intent is clearer with the for-version)

**Comment by otavioschwanck** [comment]
Very important PR!  Waiting for it

**Review by leiserfg — APPROVED** [review]

**Comment by L3MON4D3** [comment]
Thx for taking a look @leiserfg :)

**Pull request merged**